### PR TITLE
XW-3303 | autoplay toggle - release PR

### PR DESCRIPTION
**Pull request by latata**
https://wikia-inc.atlassian.net/browse/XW-3303
@Wikia/x-wing

### CODE REVIEW

File: extensions/wikia/ArticleVideo/scripts/ooyala-player.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ define('ooyala-player', function () {
 	// TODO ooyala only supports font icons so we probably need to extract our DS icons to font
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/scripts/ooyala-player.js:4]: INFO Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

File: extensions/wikia/ArticleVideo/bower_components/skin-config/validate.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    if( ! validation_result ) {
+        var err = "(tv4) VALIDATION FAILED: " + JSON.stringify( tv4.error );
+        throw err;
+    }
+}
+
+function validate( schema_object, json_object ) {
+    validate_with_jsonschema( schema_object, json_object );
+    validate_with_tv4( schema_object, json_object );
+    console.log( "Passed." );
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/bower_components/skin-config/validate.js:36]: MAJOR Remove this logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2228)

File: extensions/wikia/ArticleVideo/bower_components/skin-config/validate.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    catch( e ) {
+        var err = "FAILED TO PARSE INPUT JSON: " + msg + ", " + e.message;
+        throw err;
+    }
+}
+
+function run( json_schema_path, json_data_path ) {
+    fs.readFile( json_schema_path, function( err, schema_buf ) {
+        if( err ) { throw err; }
+        var schema_object = json_parse( "schema input", schema_buf, 'utf8' );
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/bower_components/skin-config/validate.js:55]: CRITICAL "json_parse" expects 2 arguments, but 3 were provided. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS930)

File: extensions/wikia/ArticleVideo/bower_components/skin-config/validate.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
+var fs = require( 'fs' );
+var stripComments = require( 'strip-json-comments' )
+
+// assumes this is run via node.
+function usage( message ) {
+    if( message ) {
+        console.log( message );
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/bower_components/skin-config/validate.js:7]: MAJOR Remove this logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2228)

File: extensions/wikia/ArticleVideo/bower_components/skin-config/validate.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    }
+    console.log( "USAGE: " + process.argv[1] + " <path-to-schema> <path-to-json>" );
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/bower_components/skin-config/validate.js:9]: MAJOR Remove this logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2228)

File: extensions/wikia/ArticleVideo/bower_components/skin-config/validate.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    }
+}
+
+function run( json_schema_path, json_data_path ) {
+    fs.readFile( json_schema_path, function( err, schema_buf ) {
+        if( err ) { throw err; }
+        var schema_object = json_parse( "schema input", schema_buf, 'utf8' );
+        fs.readFile( json_data_path, function( err, json_buf ) {
+            if( err ) { throw err; }
+            var json_object = json_parse( "json input", json_buf, 'utf8' );
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/bower_components/skin-config/validate.js:58]: CRITICAL "json_parse" expects 2 arguments, but 3 were provided. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS930)

File: extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/controlBar.js [hunk 7]
```diff
@@ -11,6 +11,7 @@ var React = require('react'),
     Popover = require('../views/popover'),
     VideoQualityPanel = require('./videoQualityPanel'),
     ClosedCaptionPopover = require('./closed-caption/closedCaptionPopover'),
+    ConfigPanel = require('./configPanel'),
     Logo = require('./logo');
     Icon = require('./icon');
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/controlBar.js:16]: MAJOR Add the "let", "const" or "var" keyword to this declaration of "Icon" to make it explicit. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2703)

File: extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/controlBar.js [hunk 8]
```diff
@@ -112,13 +113,28 @@ var ControlBar = React.createClass({
...
+  handleConfigPanelClick: function() {
+    this.toggleConfigPopover();
+  },
+
+  toggleConfigPopover: function() {
+    this.props.controller.toggleConfigPanelPopover();
+  },
+
+  closeConfigPopover: function() {
+    if(this.props.controller.state.configPanelOptions.showConfigPanelPopover == true) {
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/controlBar.js:133]: MAJOR Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
MINOR Remove the literal "true" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1125)

File: extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/controlBar.js [hunk 9]
```diff
@@ -112,13 +113,28 @@ var ControlBar = React.createClass({
...
+      this.toggleConfigPopover();
+    }
+  },
+
   closeQualityPopover: function() {
     if(this.props.controller.state.videoQualityOptions.showVideoQualityPopover == true) {
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/controlBar.js:139]: MAJOR Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
MINOR Remove the literal "true" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1125)

File: extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/videoQualityPanel.js [hunk 10]
```diff
@@ -25,8 +28,13 @@ var VideoQualityPanel = React.createClass({
...
     });
     this.props.togglePopoverAction();
+    this.props.toggleVideoQualityPanel();
   },
 
+    handleBackClick: function() {
+        this.props.toggleVideoQualityPopOver();
+    },
+
   render: function() {
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/videoQualityPanel.js:38]: MAJOR Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: extensions/wikia/ArticleVideo/bower_components/html5-skin/js/controller.js [hunk 11]
```diff
@@ -10,7 +10,8 @@ var React = require('react'),
     Skin = require('./skin'),
     SkinJSON = require('../config/skin'),
     Bulk = require('bulk-require'),
-    Localization = Bulk('./config', ['languageFiles/*.json']);
+    Localization = Bulk('./config', ['languageFiles/*.json']),
+    Cookies = require('js-cookie');
 
 OO.plugin("Html5Skin", function (OO, _, $, W) {
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/bower_components/html5-skin/js/controller.js:16]: MAJOR Remove the unused function parameter "W". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: extensions/wikia/ArticleVideo/bower_components/html5-skin/js/controller.js [hunk 12]
```diff
@@ -1294,8 +1329,12 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
             "showVideoQualityPopover":this.state.videoQualityOptions.showVideoQualityPopover
           }
         });
-      if(this.state.videoQualityOptions.showVideoQualityPopover == true) {
-        this.toggleVideoQualityPopOver();
+      if(this.state.configPanelOptions.showConfigPanelPopover == true) {
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/bower_components/html5-skin/js/controller.js:1332]: MAJOR Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
MINOR Remove the literal "true" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1125)

File: extensions/wikia/ArticleVideo/bower_components/html5-skin/js/controller.js [hunk 13]
```diff
@@ -1294,8 +1329,12 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
...
+
+        this.toggleConfigPanelPopover();
+      }
+      if(this.state.configPanelOptions.showVideoQualityPanel == true) {
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/bower_components/html5-skin/js/controller.js:1336]: MAJOR Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
MINOR Remove the literal "true" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1125)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 965 issues
* ![CRITICAL][CRITICAL] 20 critical
* ![MAJOR][MAJOR] 848 major
* ![MINOR][MINOR] 34 minor
* ![INFO][INFO] 63 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [html5-skin.js#L4384](https://github.com/Wikia/app/blob/13887c69d17919a84a5693c37902186c9f5fe7de/extensions/wikia/ArticleVideo/bower_components/html5-skin/build/html5-skin.js#L4384): Change this condition so that it does not always evaluate to "true"; some subsequent code is never executed. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2583)
1. ![CRITICAL][CRITICAL] [html5-skin.js#L4388](https://github.com/Wikia/app/blob/13887c69d17919a84a5693c37902186c9f5fe7de/extensions/wikia/ArticleVideo/bower_components/html5-skin/build/html5-skin.js#L4388): Change this condition so that it does not always evaluate to "true"; some subsequent code is never executed. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2583)
1. ![CRITICAL][CRITICAL] [html5-skin.js#L4392](https://github.com/Wikia/app/blob/13887c69d17919a84a5693c37902186c9f5fe7de/extensions/wikia/ArticleVideo/bower_components/html5-skin/build/html5-skin.js#L4392): Change this condition so that it does not always evaluate to "true"; some subsequent code is never executed. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2583)
1. ![CRITICAL][CRITICAL] [html5-skin.js#L4396](https://github.com/Wikia/app/blob/13887c69d17919a84a5693c37902186c9f5fe7de/extensions/wikia/ArticleVideo/bower_components/html5-skin/build/html5-skin.js#L4396): Change this condition so that it does not always evaluate to "true"; some subsequent code is never executed. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2583)
1. ![CRITICAL][CRITICAL] [html5-skin.js#L4915](https://github.com/Wikia/app/blob/13887c69d17919a84a5693c37902186c9f5fe7de/extensions/wikia/ArticleVideo/bower_components/html5-skin/build/html5-skin.js#L4915): Remove this "===" check; it will always be false. Did you mean to use "=="? [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS3403)
1. ![CRITICAL][CRITICAL] [html5-skin.js#L8708](https://github.com/Wikia/app/blob/13887c69d17919a84a5693c37902186c9f5fe7de/extensions/wikia/ArticleVideo/bower_components/html5-skin/build/html5-skin.js#L8708): "converter" expects 0 arguments, but 2 were provided. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS930)
1. ![CRITICAL][CRITICAL] [html5-skin.js#L8770](https://github.com/Wikia/app/blob/13887c69d17919a84a5693c37902186c9f5fe7de/extensions/wikia/ArticleVideo/bower_components/html5-skin/build/html5-skin.js#L8770): Correct one of the identical sub-expressions on both sides of operator "/" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1764)
1. ![CRITICAL][CRITICAL] [html5-skin.js#L9104](https://github.com/Wikia/app/blob/13887c69d17919a84a5693c37902186c9f5fe7de/extensions/wikia/ArticleVideo/bower_components/html5-skin/build/html5-skin.js#L9104): "iteratee" expects 1 argument, but 3 were provided. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS930)
1. ![CRITICAL][CRITICAL] [html5-skin.js#L10874](https://github.com/Wikia/app/blob/13887c69d17919a84a5693c37902186c9f5fe7de/extensions/wikia/ArticleVideo/bower_components/html5-skin/build/html5-skin.js#L10874): TypeError can be thrown as "initialChildren" might be null or undefined here. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2259)
1. ![CRITICAL][CRITICAL] [html5-skin.js#L15550](https://github.com/Wikia/app/blob/13887c69d17919a84a5693c37902186c9f5fe7de/extensions/wikia/ArticleVideo/bower_components/html5-skin/build/html5-skin.js#L15550): TypeError can be thrown as "inst" might be null or undefined here. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2259)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by rybmat — APPROVED** [review]

**Review by rogatty — APPROVED** [review]

File: extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/controlBar.js [hunk 14]
```diff
@@ -112,13 +113,28 @@ var ControlBar = React.createClass({
     } else {
       this.toggleQualityPopover();
       this.closeCaptionPopover();
+      this.closeConfigPopover();
     }
   },
 
-  toggleQualityPopover: function() {
```
Comment (hakubo) [inline extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/controlBar.js:118]: I think popover is one word
https://v4-alpha.getbootstrap.com/components/popovers/

**Pull request merged**